certbot/certbot/_internal/log.py

```diff
@@ -322,15 +322,23 @@ def post_arg_parse_except_hook(exc_type, exc_value, trace, debug, log_path):
         logger.error('Exiting abnormally:', exc_info=exc_info)
     else:
         logger.debug('Exiting abnormally:', exc_info=exc_info)
+        # Use logger to print the error message to take advantage of
+        # our logger printing warnings and errors in red text.
         if issubclass(exc_type, errors.Error):
-            sys.exit(exc_value)
+            logger.error(str(exc_value))
+            sys.exit(1)
```

Comment:
Did Certbot previously sometimes exit with a more specific error code depending on the reason for the failure? (If so, people might have scripts that are assuming that.) Is `exc_value` a numeric value? If not, does `sys.exit()` just interpret it as an int after interpreting it as a bool, or something?

Reply:
Nope! `exc_value` is the exception that was thrown and is crashing Certbot.

From the [sys.exit docs](https://docs.python.org/3.8/library/sys.html#sys.exit):

> The optional argument arg can be an integer giving the exit status (defaulting to zero), or another type of object. ... If another type of object is passed, None is equivalent to passing zero, and any other object is printed to stderr and results in an exit code of 1.

Reply:
Aha! That makes sense.